CDCgov/prime-reportstream
# Inline report id filter for delivery APIs

This PR inlines fetching the report ids that have been sent.

Test Steps:
With some delivered reports locally, run

```
curl --location 'http://localhost:7071/api/v1/receivers/ignore.HL7/deliveries' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer dummy' \
--data '{
    "sort": {
        "direction": "ASC",
        "property": "number_of_tests"
    },
    "pagination": {
        "page": 1,
        "limit": 10
    },
    "filters": [

    ]
}'
```
and
```
curl --location 'http://localhost:7071/api/v1/receivers/ignore.HL7/deliveries/submitters/search' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer dummy' \
--data '{
    "sort": {
        "direction": "ASC",
        "property": "sort_id"
    },
    "pagination": {
        "page": 1,
        "limit": 10
    },
    "filters": [

    ]
}'
```
and verify that you get results.

## Changes
- Changes the base clause for the recursive query to select report ids rather than pass in an array

## Checklist

### Testing

### Process

## Linked Issues


## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
